File: backend/app/schemas/screener.py
from pydantic import BaseModel
from typing import List, Optional

class ScreenedStock(BaseModel):
    symbol: str
    name: str
    currentPrice: float
    targetPrice: Optional[float] = None
    upside: Optional[float] = None
    peRatio: Optional[float] = None
    pegRatio: Optional[float] = None
    pbRatio: Optional[float] = None
    marketCap: float
    volume: int
    averageVolume: int
    change: float
    changePercent: float
    fiftyTwoWeekLow: float
    fiftyTwoWeekHigh: float
    dividendYield: Optional[float] = None
    reasons: List[str] = []
    sector: str
    industry: str

class TopMover(BaseModel):
    symbol: str
    name: str
    currentPrice: float
    change: float
    changePercent: float
    volume: int
    marketCap: float

class ScreenerResponse(BaseModel):
    category: str
    stocks: List[ScreenedStock]
    count: int

class TopMoversResponse(BaseModel):
    category: str
    stocks: List[TopMover]
    count: int

File: frontend/src/types/screener.ts
export interface ScreenedStock {
  symbol: string
  name: string
  currentPrice: number
  targetPrice?: number
  upside?: number
  peRatio?: number
  pegRatio?: number
  pbRatio?: number
  marketCap: number
  volume: number
  averageVolume: number
  change: number
  changePercent: number
  fiftyTwoWeekLow: number
  fiftyTwoWeekHigh: number
  dividendYield?: number
  reasons: string[]
  sector: string
  industry: string
}

export interface TopMover {
  symbol: string
  name: string
  currentPrice: number
  change: number
  changePercent: number
  volume: number
  marketCap: number
}

export interface ScreenerResponse {
  category: string
  stocks: ScreenedStock[]
  count: number
}

export interface TopMoversResponse {
  category: string
  stocks: TopMover[]
  count: number
}
